app/lib/services/ecf/ecf_user_updater.rb

```diff
@@ -0,0 +1,63 @@
+module Services
+  module Ecf
+    class EcfUserUpdater
+      def self.call(user:)
+        new(user:).call
+      end
+
+      attr_reader :user
+
+      def initialize(user:)
+        @user = user
+      end
+
+      def call
+        remote = user.ecf_user
+
+        # JsonApiClient::Resource uses errors for flow control, so failed saves
+        # will divert to the rescue block below
+        # I'd prefer to use the return value of save, but that's not possible
+        remote.email = user.email
+        remote.full_name = user.full_name
+        remote.get_an_identity_id = user.get_an_identity_id
```

Comment:
Would this record the errors if the remote is not about to save just one of the fields and not the last one?

Reply:
Can you clarify what you mean? The API is an all or nothing update, it won't update anything unless the record is valid.

Reply:
Yeah i think i miss read this as was trying to workout when the update happens. this made it look like it updates each field at a time. so i think this is all good

Reply:
Ah, it originally did them all together in an update() call but if nothing changed it'd hit the ECF API and cause an error since the JSON-API client would send an empty body. I had to move to this so I could check if an update was required before sending one.